### PR TITLE
add edition field for the version information

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
Signed-off-by: zhangjinpeng1987 <zhangjinpeng@pingcap.com>

### What problem does this PR solve?

Add the edition field for the version information.

### What is changed and how it works?

Add the edition field for the version information.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- No code

Side effects

- No

### Release note 

- No release note

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -66,15 +66,18 @@ pub mod storage;
 /// Returns the tikv version information.
 pub fn tikv_version_info() -> String {
     let fallback = "Unknown (env var does not exist when building)";
+    let edition = "Community";
```
Comment (BusyJay) [inline src/lib.rs:69]: How to define values other than community?

**Comment by lonng** [comment]
It's better to use the `EDITION` environment variable name. https://github.com/pingcap/tidb/pull/17170

**Comment by zhangjinpeng1987** [comment]
> It's better to use the `EDITION` environment variable name. [pingcap/tidb#17170](https://github.com/pingcap/tidb/pull/17170)

I think it's better to keep the existing style.

**Pull request merged**